### PR TITLE
Add 'Intro to Notion API' example

### PR DESCRIPTION
**Pull request by jessmitch42**
Adds a new example to the `/examples` directory. There was a request to have more code samples that build up in complexity so each section has a simple version that gets progressively more complex. 
There will be an `advanced` section added in a future PR (likely some existing examples moved over here to organize them a bit more). 
This can be easily expanded as more requests come in; for example, the current `/basic` files can be grouped into a `blocks` subdirectory if other "basic" examples get requested.

Note: Part of the request was to heavily comment this code here.

### CODE REVIEW

**Review by parulbaweja — DISMISSED** [review]
Just a couple questions, but lgtm!

File: examples/intro-to-notion-api/intermediate/3-query-database.js
Comment (jessmitch42) [reply examples/intro-to-notion-api/intermediate/3-query-database.js:99]: sure thing, just added :)

File: examples/intro-to-notion-api/README.md
Comment (jessmitch42) [reply examples/intro-to-notion-api/README.md:64]: it is!

**Comment by jessmitch42** [comment]
Hey @parulbaweja thanks for the review. I added your suggestion so it looks like it needs another review. If everything looks good, do you mind merging it too after approving? I don't have permissions to merge in this repo 🥲

**Review by parulbaweja — APPROVED** [review]

**Pull request merged**
